src/pybind/mgr/cephadm/services/monitoring.py

```diff
@@ -445,16 +447,21 @@ def generate_config(
         FSID = self.mgr._cluster_fsid
 
         clusters_credentials = {}
-        multi_cluster_config_raw = str(self.mgr.get_module_option_ex('dashboard', 'MULTICLUSTER_CONFIG'))
-        multi_cluster_config_str = multi_cluster_config_raw.replace("'", '"')
-        valid_multi_cluster_config_str = multi_cluster_config_str.replace('True', '"True"').replace('False', '"False"')
-        multi_cluster_config = json.loads(valid_multi_cluster_config_str)
-        if multi_cluster_config:
-            for url in targets:
-                credentials = self.find_prometheus_credentials(multi_cluster_config, url)
-                if credentials:
-                    clusters_credentials[url] = credentials
-                    clusters_credentials[url]['cert_file_name'] = ''
+        multi_cluster_config_str = str(self.mgr.get_module_option_ex('dashboard', 'MULTICLUSTER_CONFIG'))
+        try:
+            multi_cluster_config = literal_eval(multi_cluster_config_str)  # convert python string to dict
+        except ValueError:
+            multi_cluster_config = None
+            logger.error("Invalid JSON format for multi-cluster config")
+        try:
+            if multi_cluster_config:
+                for url in targets:
+                    credentials = self.find_prometheus_credentials(multi_cluster_config, url)
+                    if credentials:
+                        clusters_credentials[url] = credentials
+                        clusters_credentials[url]['cert_file_name'] = ''
+        except json.JSONDecodeError as e:
```

Comment:
where is the json loads that could cause this exception?

Reply:
Makes sense, removed